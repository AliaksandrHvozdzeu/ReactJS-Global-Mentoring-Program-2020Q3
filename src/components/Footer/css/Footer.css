footer {
    background-color: #424242;
    color: var(--red);
    height: 7vh;
    font-size: 1.5rem;
    position: sticky;
    z-index: 5;
}

.app-title {
    font-weight: 200;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%,-50%);
}

.app-title-name {
    font-weight: 800;
}
