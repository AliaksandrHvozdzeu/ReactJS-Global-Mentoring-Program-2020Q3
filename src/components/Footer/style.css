footer {
    background-color: #424242;
    color: var(--red);
    height: 7vh;
    font-size: 1.5rem;
    position: sticky;
    bottom: 0;
}

.app-title {
    font-weight: 200;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%,-50%);
}

.app-title-name {
    font-weight: 800;
}
