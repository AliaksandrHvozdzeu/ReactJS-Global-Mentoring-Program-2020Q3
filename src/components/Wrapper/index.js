import Wrapper from './js/Wrapper';

export default Wrapper;
