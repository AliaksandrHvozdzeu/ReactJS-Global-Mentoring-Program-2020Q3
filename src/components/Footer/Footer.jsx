import React from 'react';
import './style.css';
import Wrapper from '../Wrapper';

export default function Footer() {

  return (
    <>
      <footer>
        <Wrapper>
          <p className="app-title">
            <span className="app-title-name">netflix</span>
            roulette
          </p>
        </Wrapper>
      </footer>
    </>
  );

}
