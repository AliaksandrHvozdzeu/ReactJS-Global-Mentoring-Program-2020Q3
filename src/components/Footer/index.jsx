import React from 'react';
import './style.css';
import Wrapper from '../Wrapper';

const Footer = () => (
  <footer>
    <Wrapper>
      <p className="app-title">
        <span className="app-title-name">netflix</span>
        roulette
      </p>
    </Wrapper>
  </footer>
)

export default Footer;
