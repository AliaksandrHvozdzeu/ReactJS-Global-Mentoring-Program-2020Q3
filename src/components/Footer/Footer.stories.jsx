import React from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import Footer from './js/Footer';
import '../../App.css';

export default {
  title: 'Footer',
  decorators: [withKnobs],
};

export const Default = () => (
  <Footer />
);
