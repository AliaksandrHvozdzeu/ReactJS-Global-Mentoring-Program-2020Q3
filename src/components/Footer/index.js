import Footer from './js/Footer';

export default Footer;
