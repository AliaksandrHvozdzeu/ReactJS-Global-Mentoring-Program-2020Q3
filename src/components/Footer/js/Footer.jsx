import React from 'react';
import Wrapper from '../../Wrapper';
import '../css/Footer.css';

export default function Footer() {

  return (
    <>
      <footer>
        <Wrapper>
          <p className="app-title">
            <span className="app-title-name">netflix</span>
            roulette
          </p>
        </Wrapper>
      </footer>
    </>
  );

}
