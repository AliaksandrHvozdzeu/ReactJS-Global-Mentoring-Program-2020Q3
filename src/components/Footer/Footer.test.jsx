import React from 'react';
import rerender from 'react-test-renderer';
import { render } from 'enzyme';
import Footer from './js/Footer';
import Wrapper from '../Wrapper';

describe('when Footer rendered', () => {
  test('then Brand component is rendered', () => {
    const brandChildComponent = (
      <Wrapper>
        <p className="app-title">
          <span className="app-title-name">netflix</span>
          roulette
        </p>
      </Wrapper>
    );
    const component = rerender.create(<Footer>{brandChildComponent}</Footer>);
    const tree = component.toJSON();
    const FooterComponent = render(<Footer>{brandChildComponent}</Footer>);
    expect(FooterComponent.find('.app-title-name')).toBeTruthy();
    expect(FooterComponent.find('.app-title-name').text()).toEqual('netflix');
    expect(tree).toMatchSnapshot();
  });
});
