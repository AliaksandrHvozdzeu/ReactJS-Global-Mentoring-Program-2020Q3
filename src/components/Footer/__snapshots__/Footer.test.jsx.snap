// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Footer rendered then Brand component is rendered 1`] = `
<footer>
  <div
    className="wrapper unselectable"
  >
    <p
      className="app-title"
    >
      <span
        className="app-title-name"
      >
        netflix
      </span>
      roulette
    </p>
  </div>
</footer>
`;
